CONTRIBUTING.md

```diff
@@ -70,20 +71,73 @@ When you are contributing to changelog, please follow these suggestions:
 
 3. Once you have completed your work, create a Pull Request, ensuring that it meets the requirements listed below.
 
-### Testing
+### Requirements for Pull Requests
+
+* Please create Pull Requests against the `master` branch.
+* Please make sure that your code complies with [PEP8](https://www.python.org/dev/peps/pep-0008/).
+* One line should not contain more than 100 characters.
+* Make sure that new code is covered by a test case (new or existing one).
+* We don't like [spaghetti code](https://en.wikipedia.org/wiki/Spaghetti_code).
+* All the tests in the test-suite have to pass.
+
+## Development Environment
+
+### Requirements
+
+Currently, the development environment can be set up on __Linux__, and __MacOS__.
+Before beginning the setup, please ensure you have installed and set up the following on your machine:
+
+- Docker ([installation guide](https://docs.docker.com/install/))
+- make utility
+
+### Building a test image
+
+Open up a terminal and go to the cloned `release-bot` git repository.
+Execute the command
+
+```
+make image-test
+```
+
+It builds a Docker image using your local repository, with release-bot installed in it. This may take some time when building for the first time.
+
+To check if the build was successful, run  `docker images`. If the build was successful, an image named `release-bot-tests` will be present in the output list.
+
+### Testing new changes
+
+To test any new changes you made in the code, first build a test image using the above instructions, and then execute the following command
+
+```
+docker run -it -v path/to/conf.yaml:/home/test-user/conf.yaml release-bot-tests /bin/bash
+```
+
+* `path/to/conf.yaml` is the absolute path of the configuration file in your machine.
+
+A terminal in the container will open up. Now run the bot using the command 
+
+```
+release-bot -d -c  conf.yaml
+```
+
+You will have to build the test image every time you make new changes that need to be tested.
```

Comment:
This can be easily fixed by mounting the sources inside the container, or not invoking a container at all and running everything directly on host.

Reply:
But even after mounting, the user will have to perform a `pip install` right? And the docs recommend to use Docker containers, so should I mention running everything on the host?

Reply:
No, you don't need to run `pip install` every time if you use `PYTHONPATH` and point it to a directory with the source.

Reply:
This is how we are doing in one of our present projects: https://github.com/user-cont/conu/blob/master/Makefile#L67

We mount sources to /src inside the container and work from there.